apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -106,6 +125,63 @@ export class RunJob {
         throw new PlatformException(`Notification with id ${job._notificationId} not found`);
       }
 
+      const isSubscribersScheduleEnabled = await this.featureFlagsService.getFlag({
+        key: FeatureFlagsKeysEnum.IS_SUBSCRIBERS_SCHEDULE_ENABLED,
+        defaultValue: false,
+        organization: { _id: job._organizationId },
+        environment: { _id: job._environmentId },
+      });
+
+      if (isSubscribersScheduleEnabled && !this.shouldSkipScheduleCheck(job, notification)) {
+        const schedule = await this.getSubscriberSchedule.execute(
+          GetSubscriberScheduleCommand.create({
+            environmentId: job._environmentId,
+            organizationId: job._organizationId,
+            _subscriberId: job._subscriberId,
+          })
+        );
+
+        const subscriber = await this.subscriberRepository.findOne(
+          {
+            _id: job._subscriberId,
+            _environmentId: job._environmentId,
+            _organizationId: job._organizationId,
+          },
+          'timezone',
+          { readPreference: 'secondaryPreferred' }
+        );
```

Comment:
Just a general note, if the subscriber is refetched  again in the `schedule` usecase, or soemwhere up the scope, better to reuse it to avoid a duplicate db call if possible. (not sure if it's the case here)